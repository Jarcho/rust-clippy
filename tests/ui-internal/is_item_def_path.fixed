// run-rustfix
// aux-build:paths.rs
#![deny(clippy::internal)]
#![feature(rustc_private)]

extern crate clippy_utils;
extern crate paths;
extern crate rustc_hir;
extern crate rustc_lint;
extern crate rustc_middle;
extern crate rustc_span;

use clippy_utils::is_item;
use rustc_lint::LateContext;
use rustc_middle::ty::Ty;

#[allow(unused)]
use rustc_hir::LangItem;
#[allow(unused)]
use rustc_span::sym;

#[allow(unused)]
static OPTION: [&str; 3] = ["core", "option", "Option"];
#[allow(unused)]
const RESULT: &[&str] = &["core", "result", "Result"];

fn _f<'tcx>(cx: &LateContext<'tcx>, ty: Ty<'tcx>) {
    let _ = is_item(cx, ty, sym::option_type);
    let _ = is_item(cx, ty, sym::result_type);
    let _ = is_item(cx, ty, sym::result_type);

    #[allow(unused)]
    let rc_path = &["alloc", "rc", "Rc"];
    let _ = clippy_utils::is_item(cx, ty, sym::Rc);

    let _ = is_item(cx, ty, sym::option_type);
    let _ = is_item(cx, ty, sym::result_type);

    let _ = is_item(cx, ty, LangItem::OwnedBox);
    let _ = is_item(cx, ty, sym::maybe_uninit_uninit);
}

fn main() {}
