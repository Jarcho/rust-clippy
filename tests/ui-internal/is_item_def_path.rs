// run-rustfix
// aux-build:paths.rs
#![deny(clippy::internal)]
#![feature(rustc_private)]

extern crate clippy_utils;
extern crate paths;
extern crate rustc_hir;
extern crate rustc_lint;
extern crate rustc_middle;
extern crate rustc_span;

use clippy_utils::is_item;
use rustc_lint::LateContext;
use rustc_middle::ty::Ty;

#[allow(unused)]
use rustc_hir::LangItem;
#[allow(unused)]
use rustc_span::sym;

#[allow(unused)]
static OPTION: [&str; 3] = ["core", "option", "Option"];
#[allow(unused)]
const RESULT: &[&str] = &["core", "result", "Result"];

fn _f<'tcx>(cx: &LateContext<'tcx>, ty: Ty<'tcx>) {
    let _ = is_item(cx, ty, &OPTION);
    let _ = is_item(cx, ty, RESULT);
    let _ = is_item(cx, ty, &["core", "result", "Result"]);

    #[allow(unused)]
    let rc_path = &["alloc", "rc", "Rc"];
    let _ = clippy_utils::is_item(cx, ty, rc_path);

    let _ = is_item(cx, ty, &paths::OPTION);
    let _ = is_item(cx, ty, paths::RESULT);

    let _ = is_item(cx, ty, &["alloc", "boxed", "Box"]);
    let _ = is_item(cx, ty, &["core", "mem", "maybe_uninit", "MaybeUninit", "uninit"]);
}

fn main() {}
