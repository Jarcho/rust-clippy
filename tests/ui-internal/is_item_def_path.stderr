error: usage of `clippy_utils::is_item()` with a path to a diagnostic item
  --> $DIR/is_item_def_path.rs:28:13
   |
LL |     let _ = is_item(cx, ty, &OPTION);
   |             ^^^^^^^^^^^^^^^^-------^
   |                             |
   |                             help: try: `sym::option_type`
   |
note: the lint level is defined here
  --> $DIR/is_item_def_path.rs:3:9
   |
LL | #![deny(clippy::internal)]
   |         ^^^^^^^^^^^^^^^^
   = note: `#[deny(clippy::is_item_def_path_on_diagnostic_or_lang_item)]` implied by `#[deny(clippy::internal)]`

error: usage of `clippy_utils::is_item()` with a path to a diagnostic item
  --> $DIR/is_item_def_path.rs:29:13
   |
LL |     let _ = is_item(cx, ty, RESULT);
   |             ^^^^^^^^^^^^^^^^------^
   |                             |
   |                             help: try: `sym::result_type`

error: usage of `clippy_utils::is_item()` with a path to a diagnostic item
  --> $DIR/is_item_def_path.rs:30:13
   |
LL |     let _ = is_item(cx, ty, &["core", "result", "Result"]);
   |             ^^^^^^^^^^^^^^^^-----------------------------^
   |                             |
   |                             help: try: `sym::result_type`

error: usage of `clippy_utils::is_item()` with a path to a diagnostic item
  --> $DIR/is_item_def_path.rs:34:13
   |
LL |     let _ = clippy_utils::is_item(cx, ty, rc_path);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-------^
   |                                           |
   |                                           help: try: `sym::Rc`

error: usage of `clippy_utils::is_item()` with a path to a diagnostic item
  --> $DIR/is_item_def_path.rs:36:13
   |
LL |     let _ = is_item(cx, ty, &paths::OPTION);
   |             ^^^^^^^^^^^^^^^^--------------^
   |                             |
   |                             help: try: `sym::option_type`

error: usage of `clippy_utils::is_item()` with a path to a diagnostic item
  --> $DIR/is_item_def_path.rs:37:13
   |
LL |     let _ = is_item(cx, ty, paths::RESULT);
   |             ^^^^^^^^^^^^^^^^-------------^
   |                             |
   |                             help: try: `sym::result_type`

error: usage of `clippy_utils::is_item()` with a path to a lang item
  --> $DIR/is_item_def_path.rs:39:13
   |
LL |     let _ = is_item(cx, ty, &["alloc", "boxed", "Box"]);
   |             ^^^^^^^^^^^^^^^^--------------------------^
   |                             |
   |                             help: try: `LangItem::OwnedBox`

error: usage of `clippy_utils::is_item()` with a path to a diagnostic item
  --> $DIR/is_item_def_path.rs:40:13
   |
LL |     let _ = is_item(cx, ty, &["core", "mem", "maybe_uninit", "MaybeUninit", "uninit"]);
   |             ^^^^^^^^^^^^^^^^---------------------------------------------------------^
   |                             |
   |                             help: try: `sym::maybe_uninit_uninit`

error: aborting due to 8 previous errors

